py/desispec/desi_proc_dashboard.py

```diff
@@ -0,0 +1,383 @@
+
+    """
+
+    def __init__(self):
+        ############
+        ## Input ###
+        ############
+        parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+        parser = self._init_parser(parser)
+        args = parser.parse_args()
+        if not os.getenv('DESI_SPECTRO_REDUX'):
+            os.environ['DESI_SPECTRO_REDUX']='/global/project/projectdirs/desi/spectro/redux/'
+            os.environ['DESI_SPECTRO_DATA']='/global/project/projectdirs/desi/spectro/data/'
+        if args.nights=='all':
+            nights=listdir(os.getenv('DESI_SPECTRO_REDUX')+'/daily/exposures/')
+            nights=[int(x) for x in nights]
+        else:
+            try:
+                loc=locals()
+                cmd='nights='+args.nights
+                a=exec(cmd)
+                nights=loc['nights']
+            except:
+                nights=[]
+
+        tonight=self.what_night_is_it()
+        if not tonight in nights:
+            nights.append(tonight)
+        nights.sort(reverse=True)
+        if int(args.n_night)<=len(nights):
+            nights=nights[0:int(args.n_night)-1]
+
+        prod_dir=args.prod_dir # base directory of product
+        self.output_dir=args.output_dir # Portal directory for output html files
+        self.output_url=args.output_url # corresponding URL
+
+        ######################################################################
+        ## sub directories. Should not change if generated by the same code ##
+        ## that follows the same directory strucure ##
+        ######################################################################
+        self.data_dir=prod_dir+"/spectro/data/"
+        self.log_dir=prod_dir+"/spectro/redux/daily/run/logs"
+        self.redux_dir=prod_dir+"/spectro/redux/daily"
+
+        strTable=self._initialize_page()
+
+        for night in nights:
+            #print(night)
+            stat_night=self.calculate_one_night(night)
+            ####################################
+            #### Table for individual night ####
+            ####################################
+            strTable=strTable+self._add_html_table(stat_night,str(night))
+            
+        timestamp=time.strftime("%Y-%m-%d %H:%M:%S",time.localtime())
+        #print(timestamp)
+        running=self.check_running()
+        strTable=strTable+"<div style='color:#00FF00'>"+timestamp+" "+"desi_dailyproc running: "+running+"</div>"
+        strTable=strTable+self._add_js_script1()
+        strTable=strTable+"</html>"
+        hs=open(self.output_dir+"desi_proc_dashboard.html",'w')
```

Comment:
use `os.path.join` or otherwise include a `/` so that `output_dir` doesn't have to include a `/` as part of the user input.

Reply:
Fixed.